duneana/CAFMaker/CAFMaker_module.cc

```
@@ -172,302 +199,461 @@ namespace caf {
-          abort();
-        }
+  void CAFMaker::FillDirectionInfo(caf::SRDirectionBranch &dirBranch, const art::Event &evt) const
+  {
+    art::Handle<dune::AngularRecoOutput> dirReco = evt.getHandle<dune::AngularRecoOutput>(fDirectionRecoLabelNumu);
+    if(!dirReco.failedToGet()){
+      dirBranch.lngtrk.SetX(dirReco->fRecoDirection.X());
+      dirBranch.lngtrk.SetY(dirReco->fRecoDirection.Y());
+      dirBranch.lngtrk.SetZ(dirReco->fRecoDirection.Z());
+    }
+    else{
+      std::cout << "Warning: No AngularRecoOutput found with label '" << fDirectionRecoLabelNumu << "'" << std::endl;
+    }
 
-        // The iteration order here is the same as how we filled SRGlobal, so
-        // no need to do any work to make sure they align.
-        //
-        // NB this will all go wrong if we ever support more than one MCTruth
-        // per event.
-        for(const std::vector<systtools::VarAndCVResponse>& resp: *syst_resp){
-          for(const systtools::VarAndCVResponse& it: resp){
-            // Need begin/end to convert double to float
-            sr.xsSyst_wgt.emplace_back(it.responses.begin(), it.responses.end());
-            sr.cvwgt.push_back(it.CV_response);
-            sr.total_xsSyst_cv_wgt *= it.CV_response;
-          }
-        }
+    dirReco = evt.getHandle<dune::AngularRecoOutput>(fDirectionRecoLabelNue);
+    if(!dirReco.failedToGet()){
+      dirBranch.heshw.SetX(dirReco->fRecoDirection.X());
+      dirBranch.heshw.SetY(dirReco->fRecoDirection.Y());
+      dirBranch.heshw.SetZ(dirReco->fRecoDirection.Z());
+    }
+    else{
+      std::cout << "Warning: No AngularRecoOutput found with label '" << fDirectionRecoLabelNue << "'" << std::endl;
+    }
+  }
+
+  //------------------------------------------------------------------------------
+
+  void CAFMaker::FillEnergyInfo(caf::SRNeutrinoEnergyBranch &ErecBranch, const art::Event &evt) const
+  {
+    //Filling the reg CNN results
+    art::InputTag itag(fRegCNNLabel, "regcnnresult");
```

Avoid default values as they can hide fcl mistakes